gpt_index/indices/keyword_table/retrievers.py

```diff
@@ -120,12 +121,25 @@ class KeywordTableGPTRetriever(BaseKeywordTableRetriever):
 
     def _get_keywords(self, query_str: str) -> List[str]:
         """Extract keywords."""
-        response, _ = self._service_context.llm_predictor.predict(
+        event_id = self._service_context.callback_manager.on_event_start(
```

Comment:
same question: why do we put this hook into llm predictor?

Reply:
(see above response)